docs/docs/remark-ast.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Manipulating Remark ASTs
+---
+
+gatsby-transformer-remark empowers developers to translate Markdown into HTML to be consumed throughout Gatsby pages. Blogs and other content based sites can highly benefit from this so the code and content is separated. With this, people who will write content for the site don't need to worry about how the the site was written, but rather can focus just on writing the Markdown.
+
+In certain instances, a developer may want to customize certain Markdown nodes while being parsed. Examples could include [add syntax highlighting](/packages/gatsby-remark-prismjs/), [parse images](/packages/gatsby-remark-images), [embed videos](/packages/gatsby-remark-embed-video) and plenty of others. Throughout all of these instances, a plugin will examine the Markdown Abstract Syntax Tree (AST) and manipulate content based on certain node types or content in particular nodes.
+
+## What you will learn in this tutorial
+
+- How to learn how to understand the remark AST
+- How to integrate plugins with `gatsby-transformer-remark`
+- How to manipulate the remark AST to add additional functionality
+
+## Understanding the Abstract Syntax Tree
+
+To get an understanding at what is available in the Markdown AST, take a look at the markdown ast spec that is used in remark and other unist projects: [syntax-tree/mdast](https://github.com/syntax-tree/mdast).
+
+Starting out with a Markdown file as below:
+
+```markdown
+# Hello World!
+
+This is a [Real page](https://google.com)
+```
+
+Remark would translate this into an AST that will be available to `gatsby-transformer-remark` plugins. [AST Explorer](https://astexplorer.net/#/gist/d9029a2e8827265fbb9b190083b59d4d/3384f3ce6a3084e50043d0c8ce34628ed7477603) is a site that gives you a side-by-side view of the markdown and the outputted AST.
```

Comment:
Love the link to AST Explorer! For people who don't click through--it could be worth showing the JSON representation here, as well.

Reply:
Okay, I'll revert it so both the link and the full JSON is available

Reply:
Done